### PR TITLE
anagram: Rewrite to use hspec with fail-fast.

### PR DESCRIPTION
**Pull request by rbasso**
- Rewrite to use hspec.
- Update to match the json file.

Related to #211.

### CODE REVIEW

File: exercises/anagram/test/Tests.hs [hunk 1]
```diff
@@ -1,40 +1,140 @@
...
+               , subject     = "listen"
+               , candidates  = ["enlists", "google", "inlets", "banana"]
+               , expected    = ["inlets"]
+               }
+        , Case { description = "detects multiple anagrams"
+               , subject     = "allergy"
+               , candidates  = ["gallery", "ballerina", "regally", "clergy", "largely", "leading"]
+               , expected    = ["gallery", "regally", "largely"]
+               }
+        , Case { description = "does not detect indentical words"
```
Comment (petertseng) [inline exercises/anagram/test/Tests.hs]: I see the spelling mistake indentical is in x-common as well, remind me to file a PR there as well...

File: exercises/anagram/test/Tests.hs [hunk 2]
```diff
@@ -1,40 +1,140 @@
...
+               , subject     = "go"
+               , candidates  = ["go Go GO"]
+               , expected    = []
+               }
+        , Case { description = "anagrams must use all letters exactly once"
+               , subject     = "tapper"
+               , candidates  = ["patter"]
+               , expected    = []
+               }
+        , Case { description = "eliminates anagrams with the same checksum"
```
Comment (petertseng) [inline exercises/anagram/test/Tests.hs]: duplicate case, will also file PR in x-common

File: exercises/anagram/test/Tests.hs [hunk 3]
```diff
@@ -1,40 +1,140 @@
...
+               , subject     = "ΑΒΓ"
+               , candidates  = ["ABΓ"]
+               , expected    = []
+               }
+        , Case { description = "capital word is not own anagram"
+               , subject     = "BANANA"
+               , candidates  = ["Banana"]
+               , expected    = []
+               }
+        , Case { description = "anagrams must use all letters exactly once"
```
Comment (petertseng) [inline exercises/anagram/test/Tests.hs]: x-common has duplicate

File: exercises/anagram/test/Tests.hs [hunk 4]
```diff
@@ -1,40 +1,140 @@
...
+               , subject     = "patter"
+               , candidates  = ["tapper"]
+               , expected    = []
+               }
+        , Case { description = "accepts string arguments"
```
Comment (petertseng) [inline exercises/anagram/test/Tests.hs]: I see you got these from "string argument cases" section in x-common, I personally don't understand what they are for... maybe I need to ask. this particular case is a duplicate of an above case

**Comment by rbasso** [comment]
Thanks, @petertseng.

Please tell me as soon as you get the changes to `x-common`. Now that the test cases are separated from the code, I can use a few regexes to generate the new test suite.

**Comment by petertseng** [comment]
merged

**Pull request merged**